docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,252 @@
+
+Specify proxy settings on the command line if you are enrolling in {fleet} and
+the proxy configuration is not yet available. These settings are added to the
+`fleet.yml` file; you'll still need to
+<<fleet-proxy-configuration,configure {fleet} proxy settings>> for {fleet} to
+access the package registry and to communicate with {es}.
+
+The `enroll` and `install` commands accept the following flags:
+
+|===
+| CLI flag | Description
+
+|`--proxy-url <url>`
+|URL of the proxy server. The value may be either a complete URL or a
+`host[:port]`, in which case the http scheme is assumed.  The URL accepts optional
+username and password settings for authenticating with the proxy. For example:
+`http://<username>:<password>@<proxy host>/`.
+
+|`--proxy-disabled`
+|If specified, all proxy settings, including the `HTTP_PROXY` and `HTTPS_PROXY`
+environment variables, are ignored.
+
+|`--proxy-header <header name>=<value>`
+|Additional header to send to the proxy during CONNECT requests. Use the
+`--proxy-header` flag multiple times to add additional headers. You can use
+this setting to pass keys/tokens required for authenticating with the proxy.
+
+|===
+
+For example:
+
+[source,sh]
+----
+//TODO: Add an example here
+----
+
+NOTE: These commands require default policies to be loaded in {fleet}. Default
+policies are loaded automatically when you visit {fleet} for the first time. If
+you're not sure whether default policies are loaded, log in to {kib} and go to
+*Management > {fleet}*.
+
+[discrete]
+[[fleet-proxy-configuration]]
+== Configure {fleet} proxy settings
+
+Configure proxy settings for {fleet} when it must connect through a proxy server
+to:
+
+* Access the Elastic Package Registry
+* Communicate with {es}
+
+image::images/fleet-proxy-server.png[Image showing connections between {fleet}, {es}, and the Elastic Package Registry]
+
+TIP: If you're using our hosted {ess} on {ecloud}, you don't have to configure
+{fleet} proxy settings because they are handled by {ecloud}.
+
+[discrete]
+[[epr-proxy-setting]]
+=== Access the Elastic Package Registry through a proxy
+
+{fleet} might be unable to access the Elastic Package Registry because {kib} is
+behind a proxy server.
+
+Also your organization might have network traffic restrictions that prevent {kib}
+from reaching the public Elastic Package Registry endpoints, like
+https://epr.elastic.co/[epr.elastic.co], to download package metadata and
+content. You can route traffic to the public endpoint of EPR through a network
+gateway, then configure proxy settings in the {kib} configuration file,
+`kibana.yml`. For example:
+
+[source,yaml]
+----
+xpack.fleet.registryProxyUrl: your-nat-gateway.corp.net
+----
+
+[discrete]
+[[es-proxy-setting]]
+=== Access {es} through a proxy
+
+When {fleet} is behind a proxy server, configure {fleet} settings to
+specify the URL for the proxy server:
+
+//QUESTION: Is correct? There does not appear to be a Proxy URL setting under
```

Comment:
@michel-laterman This is another section where I have questions. Are the steps here correct? Can you provide a screen capture?

Reply:
@dedemorton Are you talking about Fleet Server here? It might be helpful to specify the components being connected to be more clear.

Reply:
Also @michel-laterman what type of proxies is this section relevant for? Would users generally configure a reverse proxy for ES and FS? If so, it might help to be more explicit. If its a forward proxy, then users would need to set `--proxy-url`? I am not an expert at this, so please correct me if I'm wrong.